src/main/java/seedu/address/logic/commands/Command.java

```diff
@@ -43,4 +43,16 @@ public void setData(Model model) {
     protected void indicateAttemptToExecuteIncorrectCommand() {
         EventsCenter.getInstance().post(new IncorrectCommandAttemptedEvent(this));
     }
+    
+    /**
+     * Does this command modify data in TaskBook? true or false
+     * @return
+     */
+    public abstract boolean modifiesData();
```

Comment:
Why is this still needed? I already gave you a solution that does not need it, and I'm strongly against adding a hodgepodge of abstract methods just to satisfy the way an algorithm does it.